### PR TITLE
Remove maxprice

### PR DESCRIPTION
**Pull request by HongjianZhang**
Before this commit, maxPrice is used to specify the upperbound for the cost
of a machine. A machine is chosen only if it fits the description and costs
less than maxPrice. However this feature is rarely used. This commit will
remove it.

### CODE REVIEW

File: blueprint/bindings.js [hunk 1]
```diff
@@ -241,14 +241,18 @@ Deployment.prototype.toQuiltRepresentation = function toQuiltRepresentation() {
 };
```
Comment (ejj) [inline blueprint/bindings.js]: In the commit message s/latter/letter

File: blueprint/bindings.js [hunk 2]
```diff
@@ -127,14 +127,12 @@ function createDeployment(deploymentOpts) {
  * @constructor
  *
  * @param {Object} deploymentOpts - An object containing options that can tweak
- *   the behavor of the namespace.  Options include: `maxPrice` which defines
- *   the price that should be bid in spot auctions for preemptible machines,
- *   `namespace` which instructs the deployment what namespace it should
```
Comment (kklin) [inline blueprint/bindings.js]: This is line-wrapped funny

File: cloud/machine/machine.go [hunk 3]
```diff
@@ -19,24 +19,23 @@ type Description struct {
...
 	case db.Vagrant:
 		return vagrantSize(ram, cpu)
 	default:
 		panic(fmt.Sprintf("Unknown Cloud Provider: %s", provider))
 	}
 }
 
-func chooseBestSize(descriptions []Description, ram, cpu blueprint.Range,
-	maxPrice float64) string {
+func chooseBestSize(descriptions []Description, ram,
```
Comment (kklin) [inline cloud/machine/machine.go]: That line break looks a little short, could you place this entire function header on the same line? If not, breaking before the `)` would look a little cleaner IMO

File: cloud/machine/machine_test.go [hunk 4]
```diff
@@ -20,23 +20,23 @@ func TestConstraints(t *testing.T) {
...
 	checkConstraint(testDescriptions, blueprint.Range{Min: 2},
-		blueprint.Range{Min: 2}, 2, "size1")
+		blueprint.Range{Min: 2}, "size1")
 
 	// Test no match
 	checkConstraint(testDescriptions, blueprint.Range{Min: 3},
-		blueprint.Range{Min: 2}, 2, "")
+		blueprint.Range{Min: 2}, "")
 
 	// Test price too expensive
```
Comment (kklin) [inline cloud/machine/machine_test.go]: This test should just be removed

File: cloud/machine/machine_test.go [hunk 5]
```diff
@@ -45,19 +45,19 @@ func TestConstraints(t *testing.T) {
 		{Size: "size4", Price: 0.5, RAM: 3, CPU: 4},
 	}
 	checkConstraint(testDescriptions, blueprint.Range{Min: 4},
-		blueprint.Range{Min: 3}, 2, "size3")
+		blueprint.Range{Min: 3}, "size3")
 
 	// Test infinite price
```
Comment (kklin) [inline cloud/machine/machine_test.go]: This test should be removed

File: engine/engine.go [hunk 6]
```diff
@@ -108,8 +106,7 @@ func toDBMachine(machines []blueprint.Machine, maxPrice float64,
 
 func machineTxn(view db.Database, bp blueprint.Blueprint, adminKey string) {
 	// XXX: How best to deal with machines that don't specify enough information?
-	maxPrice := bp.MaxPrice
```
Comment (kklin) [inline engine/engine.go]: The `MaxPrice` field can be removed from `blueprint/blueprint.go` (I found this by running `git grep -i maxprice`)

File: cloud/machine/machine_test.go (hunk 4 above)
Comment (HongjianZhang) [reply cloud/machine/machine_test.go]: The test to be removed should be "// Test price too expensive" not "Test no match" right?

Comment (kklin) [reply cloud/machine/machine_test.go]: Yep

**Comment by HongjianZhang** [comment]
Fixed comments and rebased against master.

File: blueprint/bindings.js (hunk 1 above)
Comment (HongjianZhang) [reply blueprint/bindings.js]: I don't think I have latter in this commit. I removed the previous commit that was wrongfully added to this pull request.

**Pull request merged**